src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -1816,6 +1826,10 @@ void MergeTreeData::loadDataParts(bool skip_sanity_checks, std::optional<std::un
     calculateColumnAndSecondaryIndexSizesImpl();
 
     PartLoadingTreeNodes unloaded_parts;
+
+    for (const auto & [info, name, disk] : unexpected_parts_to_load)
+        unloaded_parts.push_back(std::make_shared<PartLoadingTree::Node>(info, name, disk));
```

Comment:
Currently, we detach unexpected parts, but after this change, we will remove them like outdated parts. Should we add them to `broken_parts_to_detach` instead? (also it would be nice to detach them with `ignored_` prefix (currently we detach all `broken_parts_to_detach` with `broken-on-start_`))